### PR TITLE
Improve PdoBackend

### PR DESCRIPTION
**Pull request by bwaidelich**
Related: #1763

### CODE REVIEW

**Comment by Sebobo** [comment]
This should also replace the table names with the ones from the config.

File: Neos.Cache/Resources/Private/DDL_pgsql.sql [hunk 1]
```diff
@@ -0,0 +1,22 @@
+BEGIN;
+
+CREATE TABLE IF NOT EXISTS "cache" (
+  "identifier" VARCHAR(250) NOT NULL,
+  "cache" VARCHAR(250) NOT NULL,
+  "context" VARCHAR(150) NOT NULL,
+  "created" INTEGER NOT NULL,
```
Comment (Sebobo) [inline Neos.Cache/Resources/Private/DDL_pgsql.sql]: I saw a recommendation to use `"created" INTEGER CHECK(created >= 0) NOT NULL,` to replace the `UNSIGNED` config.

File: Neos.Cache/Resources/Private/DDL_pgsql.sql [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  "lifetime" INTEGER DEFAULT '0' NOT NULL,
```
Comment (Sebobo) [inline Neos.Cache/Resources/Private/DDL_pgsql.sql]:
```suggestion
  "lifetime" INTEGER CHECK(lifetime >= 0) DEFAULT '0' NOT NULL,
```

File: Neos.Cache/Classes/Backend/PdoBackend.php [hunk 3]
```diff
@@ -386,8 +385,12 @@ protected function connect()
     protected function createCacheTables()
     {
         $this->connect();
+        $sqlPath = __DIR__ . '/../../Resources/Private/DDL_' . strtolower($this->pdoDriver) . '.sql';
+        if (!file_exists($sqlPath)) {
+            throw new Exception(sprintf('Could not find SQL for pdo driver "%s" (at "%s")', $this->pdoDriver, $sqlPath), 1568133488);
+        }
         try {
-            PdoHelper::importSql($this->databaseHandle, $this->pdoDriver, __DIR__ . '/../../Resources/Private/DDL.sql');
+            $this->databaseHandle->exec(file_get_contents($sqlPath));
```
Comment (Sebobo) [inline Neos.Cache/Classes/Backend/PdoBackend.php]: We could first read the sql and then replace the table names. This way the method works with any number of configured caches.

File: Neos.Cache/Resources/Private/DDL_pgsql.sql (hunk 1 above)
Comment (bwaidelich) [reply Neos.Cache/Resources/Private/DDL_pgsql.sql:7]: Sorry for the (very) late reaction...
what does this do and is this syntax compatible with all supported platforms?

Comment (bwaidelich) [reply Neos.Cache/Resources/Private/DDL_pgsql.sql:7]: nevermind – this PR is totally outdated

**Comment by bwaidelich** [comment]
=> #1766

**Pull request closed without merging**